### ISSUE TITLE
Extreme lag and freezes from having a system with a lot of items

### ISSUE THREAD

**Issue by FrizouWasThere** [blushiemagic/MagicStorage#173]
If the system has a lot of items, any action in the crafting interface will range from a 1.5 second freeze (pretty much any craft) to a crash (usually only happens if you craft a lot of items with CTRL)
This doesn't seem to be caused by any other mod from what i tried

**Comment by ExterminatorX99** [comment]
Attempted fix uploaded.
Did it work?

**Comment by FrizouWasThere** [comment]
lemme check

**Comment by FrizouWasThere** [comment]
Seems to have made it worse

**Comment by ExterminatorX99** [comment]
Are you on the latest Magic Storage version? `0.5.5.1`

**Comment by NicoVirtue** [comment]
having this issue in my multiplayer world. we have 3775/4000 items. on mod version v0.5.6.5

**Comment by ghost** [comment]
Can confirm I have this issue when crafting stacks of items. Mod version 0.5.6.5

**Comment by maxsupermanhd** [comment]
MagicStorage 0.5.7.7. This lag is extremely annoying, I really really want it fixed. Crafting interface renders game useless for good 2-5 seconds. More than this, it does not queue any inputs while it lags so I can not input anything. Same goes for search - when erasing character by character I can not just hit backspace 3 times for example, I have to hit it once and wait for it to process everything and then hit backspace again! @ExterminatorX99 Begging for optimization update

**Comment by maxsupermanhd** [comment]
One solution would be to somehow make it background and display loading spinner so it does not freeze game completely.

**Comment by ExterminatorX99** [comment]
It used to be in the background but that caused issues

**Comment by jpsenior** [comment]
Although this is a fairly old issue, it's possible this is related to recipe crafting "History".  I'm not sure what the limits of this list are, but in my modded playthrough I have been playing collectathon across thousands of items with Journey mode integration.  If I clear the history occasionally, the lag vanishes.  I wonder if at least one of these problems with lag is related to crafting history. Clearing history makes the Magic Storage interface snappy again.

There's a Magic Storage Config option to disable clearing recipe history - but I reckon that would worsen the problem as we couldn't actually empty it.

To summarize: There may be inefficient storage // lookup logic as related to recipe history. Occasionally manually clearing it seems to solve all problems for me.

**Comment by absoluteAquarian** [comment]
The following message has been copy/pasted to all issues older than 3 months.

In order to clean up how the project looks, I'll be closing all old issues.  If you believe that this issue is still relevant, please make a new issue that references this issue in its message.
